Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py

```diff
@@ -0,0 +1,666 @@
+                                    is_admin=entity.get('isAdmin'), is_external=entity.get('isExternal'),
+                                    email=entity.get('email'), identifier=entity.get('username'))
+    return readable_output
+
+
+def users_accounts_to_human_readable(users_accounts, username):
+    if not username:
+        users_accounts_readable_outputs = []
+        for entity in users_accounts:
+            users_accounts_readable_outputs.append(user_account_to_human_readable(entity))
+    else:
+        users_accounts_readable_outputs = user_account_to_human_readable(users_accounts)
+    headers = ['display_name', 'last_seen', 'is_admin', 'is_external', 'email', 'identifier']
+    human_readable = tableToMarkdown('Microsoft CAS Users And Accounts', users_accounts_readable_outputs, headers,
+                                     removeNull=True)
+    return human_readable
+
+
+def list_users_accounts_command(client, args):
+    url_suffix = '/entities/'
+    custom_filter = args.get('custom_filter')
+    arguments = assign_params(skip=args.get('skip'), limit=args.get('limit'), service=args.get('service'),
+                              instance=args.get('instance'), type=args.get('type'), username=args.get('username'),
+                              group_id=args.get('group_id'), is_admin=args.get('is_admin'), status=args.get('status'),
+                              is_external=args.get('is_external'))
+    request_data, url_suffix = build_filter_and_url_to_search_with(url_suffix, custom_filter, arguments)
+    users_accounts = client.list_users_accounts(url_suffix, request_data)
+    if users_accounts.get('data'):
+        users_accounts = users_accounts.get('data')
+    if args.get('username'):
+        users_accounts = users_accounts[0]
+    human_readable = users_accounts_to_human_readable(users_accounts, args.get('username'))
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='MicrosoftCloudAppSecurity.UsersAccounts',
+        outputs_key_field='username',
+        outputs=users_accounts
+    )
+
+
+def calculate_fetch_start_time(last_fetch, first_fetch):
+    if last_fetch is None:
+        if not first_fetch:
+            first_fetch = '3 days'
+        first_fetch_time = arg_to_timestamp(first_fetch)
+        return first_fetch_time
+    else:
+        last_fetch = int(last_fetch)
+    latest_created_time = last_fetch
+    return latest_created_time
+
+
+def get_max_result_number(max_results):
+    if not max_results:
+        return DEFAULT_INCIDENT_TO_FETCH
+    return int(max_results)
+
+
+def arrange_alert_by_incident_type(alert):
+    incident_types: Dict[str, Any] = {}
+    for entity in alert['entities']:
+        if not entity['type'] in incident_types.keys():
+            incident_types[entity['type']] = []
+        incident_types[entity['type']].append(entity)
+    alert.update(incident_types)
+    del alert['entities']
```

Comment:
let's talk about this logic, and we should consider adding a google docstring here